pkg/controller/admissionchecks/multikueue/multikueuecluster.go

```diff
@@ -426,6 +443,86 @@ func (c *clustersReconciler) Reconcile(ctx context.Context, req reconcile.Reques
 	return reconcile.Result{}, client.IgnoreNotFound(c.updateStatus(ctx, cluster, true, "Active", "Connected"))
 }
 
+// validateKubeconfig checks that the provided kubeconfig content is safe to use
+// in a MultiKueueCluster context.
+func validateKubeconfig(kubeconfig []byte) error {
```

Comment:
Shouldn't we reject the insecure KubeConfig during webhook validations?
It allows us to expose the reason why MultKueue does not work well to users.

Reply:
> This would require webhook for secrets which might not be good.

I'm not sure the reason why "not good". Exposing the information only to a condition is very inconvenient.
We have 2 approaches for more user-friendly ones.

1. Implement Webhook validations
2. Validate inside of controller (current impls) and deactivate the invalid MultiKueueCluster, then expose the deactivation information to somewhere CR